queens/iterators/control_variates_iterator.py

```diff
@@ -0,0 +1,212 @@
+        result_description (dict, opt): Desctiption of results of iterator.
+        num_samples_cvs (list(int)):    Number of samples to use for computing the expectation
+                                        value of control variable. If expectation of a control
+                                        variableis known, you can pass None as it's number of
+                                        samples.
+        samples (np.array):             Samples, on which all models are evaluated on.
+    """
+
+    @log_init_args
+    def __init__(
+        self,
+        models,
+        parameters,
+        global_settings,
+        seed,
+        num_samples,
+        expectation_cvs=None,
+        result_description=None,
+        num_samples_cvs=None,
+        use_optn=False,
+        models_cost=None,
+    ):
+        """Control variates iterator constructor.
+
+        Args:
+            models (list(Model)):       Models to be used for evaluation. With models[0] being the
+                                        function that you want to approximate and models[1] being
+                                        the control variable. Pass at two models.
+            parameters (Parameters):        Parameters to use for evaluation.
+            global_settings (GlobalSettings):   Global settings to use for evaluation.
+            seed (int):                     Seed for random samples.
+            num_samples (int):              Number of samples to evaluate for estimation.
+            expecation_cvs (int)):      Here you can pass expected values of the control variate.
+                                        If the expectation value of a control
+                                        variate is not known, pass None as it's expected value.
+            result_description (dict, opt): Desctiption of results of iterator.
+            num_samples_cvs (list(int)):    Number of samples to use for computing the expectation
+                                        value of control variable. If expectation of a control
+                                        variable is known, you can pass None as it's number of
+                                        samples.
+
+        Raises:
+            ValueError:     If models is not a list of length two.
+                            If the expectation or num_samples_cvs is not given when
+                            not using optn.
+        """
+        if not isinstance(models, list):
+            raise ValueError("Models have to be given in the form of a list!")
+        if len(models) != 2:
+            raise ValueError("Two models have to be given!")
+
+        if expectation_cvs is None and num_samples_cvs is None and use_optn is False:
+            raise ValueError(
+                "expectation_cvs or num_samples_cvs has to be given, when not using optn"
+            )
+
+        if use_optn is True and models_cost is None:
```

Comment:
[PEP 8](https://peps.python.org/pep-0008/#programming-recommendations) recommends not to use `is` when working with bools. Could you please fix this in other places as well.

```suggestion
        if use_optn and models_cost is None:
```

Reply:
Fixed it! Out of curiosity, why is `is` used at all, then? I remember getting error messages when committing because I used `... == None`. Afterward, I changed many if-statements to use `is`

Reply:
When checking for `None` we use `is`:  `if blabla is None`. If `blabla` a bool,  do `if blabla`. For most other types `==` is used. See  [this](https://peps.python.org/pep-0008/#programming-recommendations)